### DISCUSSION TITLE
useFilenamingConvention add PascalCase variant

### DISCUSSION THREAD

**Discussion by marcoSven** [biomejs/biome#1409] [Ideas]
Could we add PascalCase? https://biomejs.dev/linter/rules/use-filenaming-convention/

**Comment by Conaclos** [comment]
The `export` case is not what you want? It ensures that the filename matches one of the exports (the export can be in PascalCase).

**Reply by marcoSven** [reply]
I am right now in the process of migrating our React project from ESLint. I was also anticipating that `export` case would be sufficient. Unfortunately it turns out we have not been very consistent with our exports e.g.

```tsx

export default memo(NotificationItem);

```  

We should clean these up but I was wondering if we can put this in our project backlog, since as of now we would have over 202 errors. Is there a reason PascalCase is not an option?

Thank you for the fast reply and Happy New Year! We are very excited about using Biome!

**Reply by Conaclos** [reply]
> Is there a reason PascalCase is not an option?

The idea was to model as closely as possible the needs.

However, your use-case seem legitimate. Lets me think a bit about that. We could add the variant. Not sure if we should add it in the default config.

**Reply by marcoSven** [reply]
Thank you I appreciate it. The config for our project would look like this

```json

"useFilenamingConvention": {
  "options": {
    "strictCase": false,
    "filenameCases": ["camelCase", "PascalCase"]
  }
}

```

Personally I would only expect `export` and `kebab-case` to be the default. `useFilenamingConvention` is so far the only rule that requires custom options. We are very happy with the other defaults.

**Reply by Conaclos** [reply]
The current default also adds the two others cases because it is commonly used among JS projects.

**Reply by Conaclos** [reply]
I submitted a PR #1414

**Comment by kainosnoema** [comment]
Given there are other similar explicit options like `camelCase` and `snake_case`, seems obvious that one of the most popular ones (`PascalCase`, matching component name) should be added to the list.

**Reply by Conaclos** [reply]
If you export the component, `export` already give what you want: ensuring that the filename matches the name of one of the exports.
